plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java

```diff
@@ -512,6 +517,25 @@ public void testPathColumn()
         }
     }
 
+    @Test
+    public void testTableWithNonNullableColumns()
+    {
+        String tableName = "test_table_with_non_nullable_columns_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + "(col1 INTEGER NOT NULL, col2 INTEGER, col3 INTEGER)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES(1, 10, 100)", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES(2, 20, 200)", 1);
+        assertThatThrownBy(() -> query("INSERT INTO " + tableName + " VALUES(null, 30, 300)"))
+                .hasMessageContaining("NULL value not allowed for NOT NULL column: col1");
+
+        //TODO this should fail https://github.com/trinodb/trino/issues/13434
+        assertUpdate("INSERT INTO " + tableName + " VALUES(TRY(5/0), 40, 400)", 1);
+        //TODO these 2 should fail  https://github.com/trinodb/trino/issues/13435
+        assertUpdate("UPDATE " + tableName + " SET col2 = NULL where col3 = 100", 1);
+        assertUpdate("UPDATE " + tableName + " SET col2 = TRY(5/0) where col3 = 200", 1);
```

Comment:
Could you add two test methods (e.g. `testInsertIntoNotNullColumnWithExpression` & `testUpdateNotNullColumnWithExpression`) to `BaseConnectorTest` and override them here?

Reply:
@ebyhr I can add them but they will fail there for most cases 
See https://github.com/trinodb/trino/issues/13434

Reply:
@ebyhr can we merge this one as it is and I will add sufficient tests to BaseConnectorTest in scope of https://github.com/trinodb/trino/issues/13434 and https://github.com/trinodb/trino/issues/13435 ?

Reply:
Yes, separated PR is fine to me.